course5/homework04/Stream.h

```diff
@@ -0,0 +1,41 @@
+#include <string>
+#include <cstdio>
+
+// ARS -- week 04. 20170227
+
+const std::string Endl ="\n";
+
+class Stream
+{
+private:
+	FILE*	m_fileptr = nullptr; 
+public:
+	Stream(const char* file);
+	Stream(FILE*  stream);
+	~Stream();
+	Stream& operator << (int x);
+	Stream& operator << (long l);
+	Stream& operator << (bool b);
+	Stream& operator << (char c);
+	Stream& operator << (float f);
+	Stream& operator << (double d);
+	Stream& operator << (const char* str);
+	Stream& operator << (std::string s);
+};
+
+
+/* 
+Copyable? 
+   A copy constructor for a stream is technically possible, but practically not wise, as
+   you will have 2 objects having the same file descriptor. If the one objects is destructed, the
+   other is stuck with an invalid filepointer. A possibility would be to have the copy contructor not
+   copy the filepointer, but open the file with its own filepointer or dup() the other, but again, 
+   synchronizing might become an issue
```

Comment:
Agreed. But your implementation is copyable - it has compiler-generated copy constructor.
Vide: http://en.cppreference.com/w/cpp/language/copy_constructor